internal/configs/configurator.go

```diff
@@ -265,6 +265,43 @@ func (cnf *Configurator) AddOrUpdateIngress(ingEx *IngressEx) (Warnings, error)
 	return warnings, nil
 }
 
+// GetVirtualServer takes a hostname and returns a VS for the given hostname.
+func (cnf *Configurator) GetVirtualServer(hostname string) *conf_v1.VirtualServer {
```

Comment:
nit: Could we rename this a bit more intuitively, e.g. `GetVirtualServerForHostname` or similar?